adapters/Vungle/VungleAdapter/GADMAdapterVungleInterstitial.m

```diff
@@ -56,34 +60,55 @@ - (void)dealloc {
 #pragma mark - GAD Ad Network Protocol Banner Methods (MREC)
 
 - (void)getBannerWithSize:(GADAdSize)adSize {
-  self.adapterAdType = GADMAdapterVungleAdTypeBanner;
-
   // An array of supported ad sizes.
-  NSArray *potentials = @[ NSValueFromGADAdSize(kGADAdSizeMediumRectangle) ];
+  GADAdSize shortBannerSize = GADAdSizeFromCGSize(kVNGBannerShortSize);
+  NSArray *potentials = @[NSValueFromGADAdSize(kGADAdSizeMediumRectangle), NSValueFromGADAdSize(kGADAdSizeBanner), NSValueFromGADAdSize(kGADAdSizeLeaderboard), NSValueFromGADAdSize(shortBannerSize)];
   GADAdSize closestSize = GADClosestValidSizeForAdSizes(adSize, potentials);
-  // Check if given banner size is in MREC.
+  CGSize size = CGSizeFromGADAdSize(closestSize);
+  if (size.height == kGADAdSizeBanner.size.height) {
+    if (size.width < kGADAdSizeBanner.size.width) {
+      _bannerSize = kVNGBannerShortSize;
+      self.adapterAdType = GADMAdapterVungleAdTypeShortBanner;
+    } else {
+      _bannerSize = kGADAdSizeBanner.size;
+      self.adapterAdType = GADMAdapterVungleAdTypeBanner;
+    }
+  } else if (size.height == kGADAdSizeLeaderboard.size.height) {
+    _bannerSize = kGADAdSizeLeaderboard.size;
+    self.adapterAdType = GADMAdapterVungleAdTypeLeaderboardBanner;
+  } else if (size.height == kGADAdSizeMediumRectangle.size.height) {
+    _bannerSize = kGADAdSizeMediumRectangle.size;
+    self.adapterAdType = GADMAdapterVungleAdTypeMREC;
+  } else {
+    _bannerSize = kGADAdSizeInvalid.size;
+    self.adapterAdType = GADMAdapterVungleAdTypeUnknown;
+  }
+
+  // Check if given banner size is valid
   id<GADMAdNetworkConnector> strongConnector = _connector;
-  if (!IsGADAdSizeValid(closestSize)) {
+  if (CGSizeEqualToSize(_bannerSize, kGADAdSizeInvalid.size)) {
     NSError *error = GADMAdapterVungleErrorWithCodeAndDescription(
-        kGADErrorMediationInvalidAdSize, @"Vungle only supports banner ad size in 300 x 250.");
+        kGADErrorMediationInvalidAdSize, @"Vungle only supports banner ad size in 300 x 250, 320 x 50, 300 x 50 and 728 x 90.");
     [strongConnector adapter:self didFailAd:error];
     return;
   }
 
+  VungleAdNetworkExtras *networkExtras = [strongConnector networkExtras];
   self.desiredPlacement = [GADMAdapterVungleUtils findPlacement:[strongConnector credentials]
-                                                  networkExtras:[strongConnector networkExtras]];
-
+                                                  networkExtras:networkExtras];
+  self.bannerRequest = [[GADMAdapterVungleBannerRequest alloc] initWithPlacementID:self.desiredPlacement ?: @""
+                                                                uniquePubRequestID:networkExtras.UUID ?: @""];
```

Comment:
@YueVungle we caught an issue here during testing. If 2 different banners are loaded with the same placement ID (not a refresh) and don't specify a uniquePubRequestID, both try to load but only 1 works.

Per email discussions, we agreed that by default, if uniqueRequestID isn't specified, we couldn't detect if it was a refresh and we would fail the ad.

Ultimately, I suggest not defaulting to @"" here (just take the result of networkExtras.UUID which may be nil), and changing this parameter to be nullable inside GADMAdapterVungleBannerRequest. Then when comparing GADMAdapterVungleBannerRequest objects, comparing [nil isEqualToString:nil] will fail if 2 banner requests don't specify this extra. Currently it's defaulting to a pass because [@"" isEqualToString:@""] returns YES.

This is the one remaining issue we'd like fixed prior to approvinig.

Reply:
@YueVungle Hi Eric, I know Vungle Android side already implemented above suggestion. We will implement above suggestion on iOS side too . But I have a concern about above implementation, if there is only one Banner ad in a page and doesn't register networkExtras(so uniquePubRequestID is nil), this could lead the refresh Banner request fail to load. Thanks O(∩_∩)O

Reply:
@YueVungle yes that is true, but no uniquePubRequestID also means we can't detect if it was 2 different requests. So we'd be making an assumption that it's a refresh without knowing for sure. And per our email conversation, it's a much bigger publisher experience issue to accidentally assume a refresh when it isn't, because the first ad would then be empty. So it's important to be conservative here and drop that request.

On Android, we had the adapter log a clear message when this happens, to help notify publishers of this case:

`Ad already loaded for placement ID: " + placementId + ", and cannot determine if this is a refresh. Set Vungle extras when making an ad request to support refresh on Vungle banner ads.`

Reply:
@ericleich Hi Eric, got it, thanks a lot for the detailed explanation and guidance, we will implement above suggestion. Have a great day! ^_^

Reply:
@YueVungle I think we're still waiting on getting the updated code after you finish testing. Please let me know once it's ready!

Reply:
@ericleich Hi Eric, I'm really really sorry for the late response,  because before committing the changes of this suggestion about refresh checking, the changes must pass internal PR review and QA/IE test. And QA/IE colleagues have other tests scheduled in advance, so the QA test just passed. I have updated the codes, now if uniqueRequestID isn't specified(it is set to nil), we couldn't detect if it was a refresh and we would fail the ad. Please review the changes and thanks so much for all the help and guidance. Have a great day! ^_^

Reply:
@YueVungle no worries, just wanted to make sure you weren't just waiting for me to review.

The changes are ready to merge pending final testing from @joshuajplagonera . Thanks!

Reply:
@ericleich Hi Eric, got it, big thanks to you for all the quick reply, help and guidance. I really appreciate it. 😊